pkg/rpc/client/wsclient.go

```diff
@@ -142,85 +146,90 @@ func (c *WSClient) wsReader() {
 	var connCloseErr error
 readloop:
 	for {
-		rr := new(requestResponse)
-		err := c.ws.SetReadDeadline(time.Now().Add(wsPongLimit))
-		if err != nil {
-			connCloseErr = fmt.Errorf("failed to set response read deadline: %w", err)
-			break
-		}
-		err = c.ws.ReadJSON(rr)
-		if err != nil {
-			// Timeout/connection loss/malformed response.
-			connCloseErr = fmt.Errorf("failed to read JSON response (timeout/connection loss/malformed response): %w", err)
-			break
-		}
-		if rr.RawID == nil && rr.Method != "" {
-			event, err := response.GetEventIDFromString(rr.Method)
+		select {
+		case <-c.shutdown:
```

Comment:
Do you really need this change? Closed connection will make `ReadJSON()` fail anyway.

Reply:
Fixed.